spec/02-integration/04-admin_api/04-plugins_routes_spec.lua

```diff
@@ -122,6 +124,41 @@ for _, strategy in helpers.each_strategy() do
         end)
 
 
+        describe("PUT", function()
+          -- regression test for #4191
+          it("can create a plugin", function()
+            local service = admin_api.services:insert()
+            admin_api.routes:insert({
+              paths = { "/mypath" },
+              service = { id = service.id },
+            })
+            local res = assert(client:send {
+              method = "PUT",
+              path = "/plugins/" .. utils.uuid(),
+              body = {
+                name = "key-auth",
+                service = {
+                  id = service.id,
+                }
+              },
+              headers = {["Content-Type"] = "application/json"}
+            })
+            assert.res_status(200, res)
+            local proxy_client = assert(helpers.proxy_client())
+            finally(function()
+              if proxy_client then
+                proxy_client:close()
+              end
+            end)
+            res = assert(proxy_client:send {
+              method  = "GET",
+              path    = "/mypath",
+            })
+            assert.res_status(401, res)
+          end)
```

Comment:
I have a subsequent commit locally in which I moved this test to a more appropriate test suite (plugins_map invalidations have their own test suite).